common/ast/src/parser/query.rs

```diff
@@ -285,76 +286,79 @@ pub fn order_by_expr(i: Input) -> IResult<OrderByExpr> {
     )(i)
 }
 
+#[derive(Debug, Clone)]
+pub struct SetOperationWithSpan<'a> {
+    span: Input<'a>,
+    elem: SetOperationElement<'a>,
+}
```

Comment:
We can extract it to util.rs:

```suggestion
#[derive(Debug, Clone)]
pub struct WithSpan<'a, T> {
    span: Input<'a>,
    elem: T,
}
```

Reply:
Lol,  we think the same. I am doing it